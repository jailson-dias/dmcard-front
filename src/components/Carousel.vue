<template>
  <v-carousel
    :show-arrows="false"
    :height="height"
    cycle
    hide-delimiter-background
    :interval="3000"
  >
    <v-carousel-item v-for="(item,i) in items" :key="i">
      <v-img :src="item.src" aspect-ratio="1.7" :contain="item.contain" height="100%" width="100%"></v-img>
    </v-carousel-item>
  </v-carousel>
</template>

<script>
export default {
  data() {
    return {
      items: [
        {
          src: require("@/assets/app-mockup.png"),
          contain: true
        },
        {
          src: require("@/assets/atuamos.jpg")
        },
        {
          src: require("@/assets/perguntas.jpg")
        }
      ],
      height: 0
    };
  },
  mounted() {
    this.onResize();
  },

  methods: {
    onResize() {
      this.height = window.innerHeight;
    }
  }
};
</script>

<style>
.v-carousel__controls {
  justify-content: flex-start !important;
}

.v-carousel__controls__item {
  height: 15px !important;
  width: 15px !important;
  margin-left: 1px !important;
  margin-right: 1px !important;
}

.v-carousel__controls__item .v-btn__content .v-icon {
  font-size: 10px !important;
  height: 15px !important;
  width: 15px !important;
}

.v-icon {
  color: #00000055 !important;
}
</style>