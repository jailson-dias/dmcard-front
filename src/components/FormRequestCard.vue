<template>
  <div id="request-card-form">
    <div class="d-flex flex-column">
      <h2 class="d-flex text-center display-1">Solicite o seu cartão aqui.</h2>
      <p class="text-justify">Para começar precisamos de algumas informações suas.</p>
      <Input label="Nome" />
      <Input label="CPF" />
      <Input label="E-mail" />
      <Input label="Celular" />
      <Button />
    </div>
  </div>
  <!-- <v-container id="input-usage" fluid>
    <v-row>
  <v-col cols="12">-->
  <!-- <v-input :messages="['Messages']" append-icon="close" prepend-icon="phone">Default Slot</v-input> -->
  <!-- </v-col>
    </v-row>
  </v-container>-->
  <!-- <Input label="Alguma Coisa" />
    <Input label="outra Coisa" />
  <Button />-->
</template>

<script>
import Input from "./Input";
import Button from "./Button";

export default {
  components: {
    Input,
    Button
  }
};
</script>

<style>
#request-card-form {
  padding-left: 60px !important;
  padding-right: 60px !important;
}

#request-card-form h2 {
  margin-top: 50px;
}

#request-card-form p {
  font-size: 20px;
  margin-top: 20px;
  margin-bottom: 20px;
}
</style>