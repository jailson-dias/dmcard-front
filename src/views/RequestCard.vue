<template>
  <div>
    <Carousel />
    <FormRequestCard />
  </div>
</template>

<script>
import Carousel from "../components/Carousel";
import FormRequestCard from "../components/FormRequestCard";
export default {
  components: {
    Carousel,
    FormRequestCard
  }
};
</script>

<style>
</style>