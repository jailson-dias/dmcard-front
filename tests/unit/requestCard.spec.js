import { shallowMount, createLocalVue } from "@vue/test-utils";
import RequestCard from "@/views/RequestCard.vue";
import Carousel from "@/components/Carousel.vue";
import FormRequestCard from "@/components/FormRequestCard.vue";
import Router from "vue-router";

const localVue = createLocalVue();
localVue.use(Router);

describe("App page", () => {
  const build = () => {
    const wrapper = shallowMount(RequestCard, { localVue });

    return {
      wrapper,
      Carousel: () => wrapper.find(Carousel),
      FormRequestCard: () => wrapper.find(FormRequestCard)
    };
  };

  it("É uma página do vue", () => {
    const { wrapper } = build();
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("A página tem o carousel e o form para preencher os dados", () => {
    const { Carousel, FormRequestCard } = build();

    expect(Carousel().exists()).toBe(true);
    expect(FormRequestCard().exists()).toBe(true);
  });
});
