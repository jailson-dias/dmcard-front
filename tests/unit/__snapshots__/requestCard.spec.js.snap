// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App page É uma página do vue 1`] = `
<div>
  <carousel-stub></carousel-stub>
  <formrequestcard-stub></formrequestcard-stub>
</div>
`;
